### PR TITLE
Disable AutoUpdateManager for GroupItems

### LINKED ISSUE #1330: Group switch update in HABPanel doesn't work correct
After upgrading to OH 2.5.x I’m experiencing unexpected behavior in HABPanel. After pressing on a group switch (scenario from ON to OFF), the status of the group switch is not updated until a page reload has been done. It worked flawlessly before on 2.4.x. I've documented my findings in the attached PDF file. Hopefully this can be resolved soon. Many thanks in advance!
[Group switch issue in HABPanel.pdf](https://github.com/openhab/openhab-core/files/4076482/Group.switch.issue.in.HABPanel.pdf)
 

In the following thread more users report the strange behavior in HABPanel:
[https://community.openhab.org/t/group-switch-not-updating/87407](url)

### PR DESCRIPTION
**Pull request by silamon**
Fixes #1330

### CODE REVIEW

File: bundles/org.openhab.core.thing/src/main/java/org/openhab/core/thing/internal/AutoUpdateManager.java [hunk 1]
```diff
@@ -281,14 +289,19 @@ private boolean isAcceptedState(State newState, Item item) {
             // Look for class hierarchy
             for (Class<?> state : item.getAcceptedDataTypes()) {
                 try {
-                    if (!state.isEnum() && state.newInstance().getClass().isAssignableFrom(newState.getClass())) {
+                    if (!state.isEnum() && state.getDeclaredConstructor().newInstance().getClass()
```
Comment (cweitkamp) [inline bundles/org.openhab.core.thing/src/main/java/org/openhab/core/thing/internal/AutoUpdateManager.java:292]: :+1:

File: bundles/org.openhab.core.thing/src/main/java/org/openhab/core/thing/internal/AutoUpdateManager.java [hunk 2]
```diff
@@ -281,14 +289,19 @@ private boolean isAcceptedState(State newState, Item item) {
...
                         break;
                     }
                 } catch (InstantiationException e) {
                     logger.warn("InstantiationException on {}", e.getMessage(), e); // Should never happen
                 } catch (IllegalAccessException e) {
                     logger.warn("IllegalAccessException on {}", e.getMessage(), e); // Should never happen
+                } catch (NoSuchMethodException e) {
+                    logger.warn("NoSuchMethodException on {}", e.getMessage(), e); // Should never happen
+                } catch (InvocationTargetException e) {
+                    logger.warn("InvocationTargetException on {}", e.getMessage(), e); // Should never happen
```
Comment (cweitkamp) [inline bundles/org.openhab.core.thing/src/main/java/org/openhab/core/thing/internal/AutoUpdateManager.java:304]: I guess we can shrink this block to be a one liner:

```suggestion
                } catch (InstantiationException | IllegalAccessException | NoSuchMethodException | InvocationTargetException e) {
                    logger.warn("Exception on {}", e.getMessage(), e); // Should never happen
```

**Review by cweitkamp — COMMENTED** [review]
Thanks for this change.

@openhab/core-maintainers Do we agree on auto update `Recommendation.DONT` for Group items?

**Comment by kaikreuzer** [comment]
> @openhab/core-maintainers Do we agree on auto update Recommendation.DONT for Group items?

It looks like the right thing to do for me, yes. 👍

**Review by cweitkamp — APPROVED** [review]
LGTM. Thanks.

**Pull request merged**